src/compiler/checker.ts

```diff
@@ -40010,7 +40013,12 @@ namespace ts {
             }
         }
 
-        function getNonInterhitedProperties(type: InterfaceType, baseTypes: BaseType[], properties: Symbol[]) {
+        function isPropertyAbstractOrInterface(declaration: Declaration, baseDeclarationFlags: ModifierFlags) {
+            return baseDeclarationFlags & ModifierFlags.Abstract && (!isPropertyDeclaration(declaration) || !declaration.initializer)
+                || isInterfaceDeclaration(declaration.parent);

```

Comment:
Should we only care that the declaration is an interface? I find it interesting that the following isn't *currently* an error:

```ts
declare class X1 {
    get y(): number;
}

class Y1 extends X1 {
    y = 1; // error (correct)
}


interface X2 {
    get y(): number;
}

declare var X2: {
    new (): X2;
    prototype: X2;
}

class Y2 extends X2 {
    y = 1; // no error (incorrect?)
}
```

Maybe we can be weaker when the property on the interface is just `PropertySignature` because of back-compat, but we probably *should* be erroring if the property on the interface is a `GetAccessorDeclaration` or `SetAccessorDeclaration` and you're trying to override it with a field.

Reply:
That's a good idea, but without the commit with create-unions-of-accessors-as-accessors, it adds a weird wrinkle when synthetic properties *are* all accessors: specifically, if you modify accessorsOverrideProperty9 so that all bases use accessors (as in your other comment), then the error shows up again: the base symbol is synthetic, but all 3 bases are accessors, so the exemption no longer applies.

Instead, execution falls through to the flags check, but the flags are wrong, so the error message is wrong.

It's probably more correct to change the way synthetic accessors are created, but it's a much further-reaching change, and most of this code is still needed. I'd rather have a more pressing reason than this to change the code that creates synthetic accessors.